src/plays/password-generator/PasswordGenerator.jsx

```diff
@@ -58,89 +61,107 @@ function PasswordGenerator(props) {
 
   // Generate a random password
   const generatePassword = () => {
-    setError(false);
-    const concated = arrangeData();
-    let finalPassword = "";
-    for (let i = 0; i < passwordConfig.length; i++) {
-      finalPassword += concated[randomNumberGenerator(concated.length)];
-    }
-    return finalPassword;
+      setError(false);
+      const concated = arrangeData();
+      let finalPassword = "";
+      for (let i = 0; i < passwordConfig.length; i++) {
+        finalPassword += concated[randomNumberGenerator(concated.length)];
+      }
+      return finalPassword;
   };
 
   // generate password button click handler
   const generateHander = () => {
     const finalPassword = generatePassword();
     setPassword({ status: false, password: finalPassword });
+    passwordField.current.value = finalPassword;
   };
 
-  // handling checkbox and updating the config
-  const handleCheckedItem = (name, checked) => {
+  // Copy the password to the clipboard
+  const onCopyClick = (e) => {
+    e.preventDefault();
+    navigator.clipboard.writeText(password.password);
+    setPassword({ ...password, status: true });
+  };
+
+  const ErrorBox = () => {
+    return <p className='error'>You cannot Uncheck All At Once.</p>;
+  }; 
+
+  const checkhandler = (id, inputCheckbox) => (e) => {
     const modifiedConfig = { ...passwordConfig };
     delete modifiedConfig.length;
     delete modifiedConfig.excludeSimilarCharacters;
-    modifiedConfig[name] = checked;
+    modifiedConfig[id] = inputCheckbox.current.checked;
     const values = Object.values(modifiedConfig).filter((i) => i === true);
     if (values.length === 0) {
       return setError(true);
     }
-    setPasswordConfig({ ...passwordConfig, [name]: checked });
+    setPasswordConfig({ ...passwordConfig, [id]: inputCheckbox.current.checked });
     setError(false);
-  };
-
-  // Copy the password to the clipboard
-  const onCopyClick = (e) => {
-    e.preventDefault();
-    navigator.clipboard.writeText(password.password);
-    setPassword({ ...password, status: true });
+    inputCheckbox.current.checked = !inputCheckbox.current.checked;
   };
 
   useEffect(() => {
     generateHander();
   }, []);
 
-  const ErrorBox = () => {
-    return <p className='error'>You cannot Uncheck All At Once.</p>;
+  const setLength = (value) => {
+    setPasswordConfig({ ...passwordConfig, ['length']: value });
+    setError(false);
```

Comment:
the value is a string here. although the for loop works but this needs to be a number.